shelley/chain-and-ledger/formal-spec/software-updates.tex

```diff
@@ -1,27 +1,34 @@
-\section{Software Updates}
+\section{Software Update Adoption}
 \label{sec:software-updates}
 
 This section describes how software updates are performed.
 The process of upgrading the system a new version consists of:
 
-devops specially monitor
-make the call
-
-
-
 \begin{enumerate}
-  \item New software is ready for downloading. The core nodes propose changing the
-    \textit{application version}. If accepted, the proposal
-    will change the application version at the start of the next epoch.
-  \item All nodes see a new application version and update their software.
-    Note that the new software should be compatible with the older version, and the new
-    ledger rules will not be applied as long as the protocol version in the
-    protocol parameters is not increased.
-  \item Every block producer includes the protocol version corresponding to the software
-    they are using in the blocks they produce.
-  \item The core nodes monitor how many block headers contain the new protocol version.
-    Once enough blocks have the new version, the core nodes post another protocol parameter
-    update to increas the protocol version.
-    If this proposal is accepted, it will be changed at the start of the new epoch and
-    the current software then makes use of new rules.
+  \item New software is ready for downloading.
+  \item The core nodes propose and vote on changing the
+    \textit{application version}.
+    The distinction between protocol and software updates, as well
+    as the voting and proposal of AV updates, is discussed
+    in Section \ref{sec:update}.
+  \item If accepted, the proposal is stored on the blockchain as the future AV
+    $\var{favs} = s \mapsto ap$,
+    indexed by the slot number $s$ in which $ap$
+    is scheduled to become the official version.
+  \item The users can now download the applications listed in $ap$
+  \item In slot $s$, $ap$ becomes the official version, stored as the value
+  $\var{avs}$ on the blockchain
```

Comment:
after slot $s$, the next block that comes in triggers updating the app versions. technically, it could be that there is a long enough gap that one whole version is skipped, but this is unlikely. anyway, I worry that going too deep into these details makes it confusing, and we can talk about proposals being accepted, etc.